### PR TITLE
Added checking in solutions listing in solution expose

### PR DESCRIPTION
**Pull request by AhmedSa-mir**
### Description

Added checking in solutions listing in solution expose

### Related Issues

https://github.com/threefoldtech/js-sdk/issues/729

### CODE REVIEW

File: jumpscale/packages/tfgrid_solutions/chats/exposed.py [hunk 1]
```diff
@@ -46,6 +46,8 @@ def solution_type(self):
 
     @chatflow_step(title="Solution to be exposed")
     def exposed_solution(self):
+        if len(self.sols.keys()) == 0:
```
Comment (m-motawea) [inline jumpscale/packages/tfgrid_solutions/chats/exposed.py:49]: maybe use `if not self.sols` ?

**Review by m-motawea — APPROVED** [review]

File: jumpscale/packages/tfgrid_solutions/chats/exposed.py [hunk 2]
```diff
@@ -46,6 +46,8 @@ def solution_type(self):
 
     @chatflow_step(title="Solution to be exposed")
     def exposed_solution(self):
```
Comment (xmonader) [inline jumpscale/packages/tfgrid_solutions/chats/exposed.py:48]: can we check the length of all solutions in the beginning of the chatflow (and show only a list of what we have solutions for)

**Review by xmonader — CHANGES_REQUESTED** [review]
I think we need to update how this works

1- we shouldn't start with the chatflow if we don't have solutions at all
2- we should filter out solution types that don't have any deployed instances.

**Review by xmonader — CHANGES_REQUESTED** [review]
can we change the message of 
flist to be -> generic container
minio -> s3 storage

?

**Comment by AhmedSa-mir** [comment]
> can we change the message of
> flist to be -> generic container
> minio -> s3 storage
> 
> ?

This should be a separate issue because then we need to change these fields in different places in code

**Pull request merged**